lang/csharp/src/apache/main/Reflect/ReflectArrayHelper.cs

```diff
@@ -0,0 +1,43 @@
+/*  Copyright 2019 Pitney Bowes Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Concurrent;
+
+namespace Avro.Reflect
+{
+    public class ReflectArrayHelper
```

Comment:
I'm reaching a bit here, but do you think that we could somehow indicate the type of array to create in the `AvroFieldAttribute`, rather than providing this array helper to the reader/writer, which means that all array-like structures must be represented using the same C# type?

Reply:
Great idea. Will add a new attribute for array types and allow the helper type to be specified there. Are you good with using dynamic in the standard helper - so anything with Add(), Clear() and Count() will work? There is so much dynamic code going on anyway..... The standard helper would cover everything implementing IList<> as well as ConcurrentBag<>.

Reply:
Got very close. Problem now is that Array is an Unnamed type so it isnt possible to look up the array type. I could switch it to a named type, with default name "array" like it is now. Alternatively is there some other field in the Avro spec we could use to bind the array to its C# type? What do you think?

Reply:
The spec says this: "Attributes not defined in this document are permitted as metadata, but must not affect the format of serialized data." What about adding a new attribute to array for the C# array helper binding?

Reply:
Added metadata attribute "helper"